tools/build_api.py

```diff
@@ -310,9 +312,15 @@ def prepare_toolchain(src_paths, target, toolchain_name,
 
     # Toolchain instance
     try:
-        toolchain = TOOLCHAIN_CLASSES[toolchain_name](
-            target, notify, macros, silent,
-            extra_verbose=extra_verbose, build_profile=build_profile)
+        if toolchain_name == "GCC_ARM":
```

Comment:
The build api should not have to know what compiler it is talking to. This implies that something is a leaky abstraction and should be re-thought.